### PR TITLE
[YUNIKORN-1235] pass priority on pods from shim into the core

### PR DESCRIPTION
**Pull request by manirajv06**
### What is this PR for?
Extract `Priority` from Pod spec and pass onto core via SI message

### What type of PR is it?
* [ ] - Improvement

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1235

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Review by wilfred-s — CHANGES_REQUESTED** [review]
The lint failure must be fixed.
Instead of using `assert.Assert(t, <pointer> != nil)` we should use an if structure to make it clear that you test against nil:
```
if <pointer> == nil {
    t.Fail("unexpected nil pointer")
}
```

File: pkg/common/test/recoverable_apps_mock.go [hunk 1]
```diff
@@ -19,7 +19,7 @@
 package test
 
 import (
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (pbacsko) [inline pkg/common/test/recoverable_apps_mock.go:22]: Nit: why this change?
